src/components/details/PokemonDetails.tsx

```diff
@@ -1,15 +1,53 @@
 import { StatusBar } from 'expo-status-bar';
-import { SafeAreaView, StyleSheet, Text, View } from 'react-native';
+import { ActivityIndicator, SafeAreaView, StyleSheet, Text } from 'react-native';
+import React, { useState, useEffect } from 'react';
+import { View } from 'react-native';
 
 
-export function PokemonDetails({ route, navigation }: { route: any, navigation: any }) {
+
+export function PokemonDetails({ route }: { route: any, navigation: any }) {
     const { name, uri } = route.params;
 
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const [data, setData] = useState<PokemonStats | null>(null);
+
+    const fetchDetails = async () => {
+        try {
+            const response = await fetch(uri)
+            const result = await response.json();
+
+            setData(result);
+        } catch (error) {
+            console.error('Error fetching data:', error);
+        }
+    }
+
+    useEffect(() => {
+        fetchDetails().then(() => {
+            setIsLoading(false);
+        });
+    }, []);
+
+
     return (
         <SafeAreaView style={styles.container}>
             <Text>Pokemon Details</Text>
             <Text>{name}</Text>
             <Text>{uri}</Text>
+            {isLoading ? (
+                <ActivityIndicator />
+            ) : (
+                !data ? (
```

Comment:
Nesting ifs is usually something to be avoided especially with ternarys. Won't tell you what to do but look into https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing
and the logical AND operator on here https://react.dev/learn/conditional-rendering